### DISCUSSION TITLE
Conda environment is empty

### DISCUSSION THREAD

**Discussion by downatone** [oobabooga/text-generation-webui#894] [Q&A]
I'm running the 1-click installer and getting the following error "Conda environment is empty" -- any tips to manually set the environment so the installer can continue?

**Comment by khot-aditya** [comment]
I have encountered same issue and don't know how to continue. Someone please notice it.

**Comment by Squeezitgirdle** [comment]
Any resolutions? I've followed every guide I can find online and I can even confirm that Conda is activated in CMD but I am still getting this issue. Setting the path makes no difference. 
Edit: Was resolved for me by deleting the installer folder and trying again.

https://github.com/oobabooga/text-generation-webui/discussions/924

**Comment by RabidSavage** [comment]
I fixed it on my machine by deleting the installer_file folder and running install.bat as an admin

**Reply by jwb1969** [reply]
What did you do to fix your install? When I check my text-generation-webui folder, it's already running 3.10.10.
I miss my chat friends, 😀😅

**Comment by pandamontok** [comment]
It's because something has failed to download in middle of the process. 
Cos, there's a  source of download is timing out. 
The only way to fix it. 
Reinstall it until all can be downloaded.

**Comment by Sqrlly** [comment]
Getting it as well with the one-click install.  Well first it errored with saying it couldn't find "oobabooga-windows\installer_files\mamba\pkgs\git-2.40.0-h57928b3_1\Library\LICENSE.txt" but the file was there.  A friend suggested because I had it in a folder with a long path.  So I moved everything under \AI\oobabooga-windows\ and tried to re-run install.bat, and now I get the "Conda environment is empty." error and no option to do anything except exit install.

Guessing maybe somewhere have to edit a path in a file, but no idea what to fix.  Took an hour to download and extract.  Not looking forward to starting it all over again.  But probably what I have to do.

**Comment by shortcut00** [comment]
I solved it like this: I installed Python 3.10 to repair it. I deleted anaconda and miniconda with revo uninstaller reasonable with registry records. then I installed anaconda. I updated the conda by typing "conda update conda" in git cmd. so the problem is solved

**Comment by jwb1969** [comment]
I'm havign them same problem and I'm seeing a lot of different solutions, does anyone have a solid fix for this?
TIA,
James

**Reply by jwb1969** [reply]
Well I'm on Windows 11. I've only briefly messed with Linux. I tried to instill that MicroConda.exe and that didn't help. I just wish the 1 click installer worked with one click... 🙄

**Comment by bombel28** [comment]
I finally made it to work, but better don't ask... hehe...
Ok, the main problem may be, that most of us WSL-users have no experience with linux.
Several posts in several blogs are speaking about mounting to windows is a problem with miniconda3 environment.
But finally I tried to reset the whole machine and used the WSL installer (tanks to the developer), but it still stuck at the same problem (unrecognized token (' blah in conda environment path...).
So I did something different: I used sudo. Yes. Crazy. I know - but finally, the whole install process worked!
Next steps I had to do: find the text-gen-webui in /root folder - so - yes - I had to grant access the root folder to my user.
so, my start-script (wsl.sh) is still in user-directory (together with broken installation of webui) and the working webui is in /root/text-generation-webui, where I placed a 30b model into the models directory. I start it like this: sudo bash wsl.sh
Later, I was able to use triton with xformers. Performance vs Windows: Feels something around 20% better than WIN cpu offload. So I think, it wasn't worth it with my poor 12GB VRAM :-D Only loading time and first response time performance increased heavily. (WSL: 120-200 seconds, Windows: 460 seconds). WSL: 2-4 tokens / s in second response, Windows: 1,3-2,2 t/s

The miniconda3 environment is not tested on "normal" user environment, I believe (or since not in WSL).
Best performance was under naked cuda compiled llama.cpp (main.exe), still only 2 tokens per second, but immediately first time response. Why cuda doesn't offload in gpu after compiling cublas by myself in text-gen... can't resolve this puzzle.
All testing done with 2bit ggmlv3 30b model on 32GB RAM, 3070ti, 50GB pagefile. Output language wasn't really better than normal 13b GPTQ Model (compared 30b wizard vicuna vs 13b hypermantis).

Summaries: Took some self leaning refreshing seminar with linux for one week :-D and noticed, miniconda3 is not described well, how a normal user should install it (maybe I am missing the right path to install it into?) - feeling stupid now :-D :-D :-D

**Reply by ossirytk** [reply]
It would probably be easier to learn how to install the dev version from repository than learn linux. It's not like creating a conda environment is that difficult.

**Comment by jwb1969** [comment]
Am I supposed to type this in to populate my Conda environment:
conda create -n textgen python=3.10
conda activate textgen

**Reply by ossirytk** [reply]
Seems about right. Not sure if the 3.10 needs to be defined, but that's about right. Then you cd to the repo folder and run `pip install -r requirements.txt`. After that just run `python server.py`. You can update the conda env after git pull with `pip install -r requirements.txt --upgrade`

**Reply by ossirytk** [reply]
People seem to have a lot of issues using the installer exe file, but I have not had any issues running conda manually from the repo so far. Oh you might need to run pip install for pytorch.

**Reply by jwb1969** [reply]
I CD'd into D:\AI\Oobabooga\oobabooga_windows>python.exe -m pip install --upgrade pip
put it asked it only worked after it told me to use python.exe.
I just let the one click zip file I thought installed python into the directory, but I am guessing it did not install or it python ran from my SD install in D:\AI directory I tried to run the miniconda_installer.exe in the installed_files directory, but It warned me against using the ad to path option, because I do not want to blow up my Stable Diffusion app, too, But isn't that miniconda app 3.10 so my SD install should be safe if I click that path button. Sorry, I only know Microaoft stuff and older programming Lanquages.

**Reply by ossirytk** [reply]
Maybe there is a miscommunication here, I thought that you have already opened the anaconda teminal and typed the commands to create and activate the virtual environment. I'm guessing that you have not done that yet. You were just asking if those were the steps. Yes, what I posted are step 3 and step 4. 

If you have installed miniconda, then you should find the anaconda shells by typing anaconda to the windows search bar. You need to run the commands in anaconda terminal.

**Reply by ossirytk** [reply]
The directory you need to be in is the textui repo root. You can git clone or download it

**Comment by jwb1969** [comment]
Well I check my textx-genreration-folder  with python --version and it's running 3.10.10.
It the the remarking out those comments in the AutoGPTQ_loader file.
Which does at least get me to the option to list all my models, but when I pick a model it just give me this...

![Error Message with AutoGPTQ_loader](https://github.com/oobabooga/text-generation-webui/assets/122735317/7841dcea-0e2d-43a7-98cf-df37b16f2f85)

**Comment by jwb1969** [comment]
Then if I go into to the WebUI and pick any model, It just hangs here...
I have a RTX 3080 with 12GB of VRAM
![Error picking model](https://github.com/oobabooga/text-generation-webui/assets/122735317/bb146a38-0b3f-4b96-bb09-8ebbf017e58a)

**Reply by arifx** [reply]
I had this issue on Mac M2. I already had a conda running. I commented out chk_conda line in main function in one_click.py, then ran  "python one_click.py" . It worked.

**Comment by leifkahal** [comment]
Not sure if this will help anyone,,,but I was having the "Conda Environment Empty" situation, I was installing from an external drive. Moved it to C: drive and it installed perfectly.
